mobify/platform-scaffold
# Fix and clean up navigation parsing

This PR fixes an error in the navigation parser that was preventing the checkout pages from loading fully. It also cleans up the navigation parsing code and makes it more reliable.

 **JIRA**: N/A

## Changes
- Remove the non-connector copy of the navigation parser
- Change the logout action to use `fetchPageData`
- Pass the login state into the navigation parser rather than duplicating the login check logic
- Unify the use of the `LOGGED_IN_NAV` and `GUEST_NAV` objects between the parser and reducer
- Hard-code the login path to avoid throwing exceptions on checkout pages.

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Make sure that the checkout pages load without errors.
